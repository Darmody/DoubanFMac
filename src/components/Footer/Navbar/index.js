// @flow
import React, { PureComponent } from 'react'
import styled from 'styled-components'
import {
  Radio as IconRadio,
  Playlist as IconPlaylist,
  HollowLove as IconLove,
  Search as IconSearch,
} from 'components/Icons'
import IconButton from './IconButton'

const Navbar = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  height: 6.25rem;
`

export default class NavbarComponent extends PureComponent {
  render() {
    return (
      <Navbar>
        <IconButton><IconRadio /> 兆赫</IconButton>
        <IconButton><IconPlaylist />  歌单</IconButton>
        <IconButton><IconLove />  我的</IconButton>
        <IconButton><IconSearch /> 搜索</IconButton>
      </Navbar>
    )
  }
}
