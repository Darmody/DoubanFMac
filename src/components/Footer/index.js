// @flow
import React, { PureComponent } from 'react'
import styled from 'styled-components'
import { shell } from 'electron'
import Navbar from './Navbar'

const Footer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  background-color: white;
  width: 100%;
  margin: 0 auto;
`

const Ad = styled.p`
  width: 100%;
  text-align: center;
`

const Purchase = styled.a`
  font-size: .75rem;
  font-weight: normal;
  text-align: center;
  color: rgb(151, 126, 66);
  text-decoration: none;
  cursor: pointer;
`

export default class FooterComponent extends PureComponent {
  handlePurchase = () => shell.openExternal('https://douban.fm/upgrade')

  render() {
    return (
      <Footer>
        <Navbar />
        <Ad>
          <Purchase href="javascript:void(0);" onClick={this.handlePurchase}>
            一键购买豆瓣FM PRO，高音质无广告 ➝
          </Purchase>
        </Ad>
      </Footer>
    )
  }
}
