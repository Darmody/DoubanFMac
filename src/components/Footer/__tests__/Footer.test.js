import React from 'react'
import renderer from 'react-test-renderer'
import Footer from 'components/Footer'

describe('Component::Footer', () => {
  test('渲染正确的 DOM 结构', () => {
    const component = renderer.create(<Footer />)
    const tree = component.toJSON()

    expect(tree).toMatchSnapshot()
  })
})
