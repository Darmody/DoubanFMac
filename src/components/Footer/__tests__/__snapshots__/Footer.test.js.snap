// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component::Footer 渲染正确的 DOM 结构 1`] = `
<div
  className="iOlBOX"
>
  <div
    className="finuIx"
  />
  <p
    className="hjfzBz"
  >
    <a
      className="jiANlk"
      href="javascript:void(0);"
      onClick={[Function]}
    >
      一键购买豆瓣FM PRO，高音质无广告 ➝
    </a>
  </p>
</div>
`;
